File: apps/chat-android/messaging-websocket/src/main/java/com/amazonaws/services/chime/sdk/messaging/internal/WebSocketAdapterObserver.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package com.amazonaws.services.chime.sdk.messaging.internal

import com.amazonaws.services.chime.sdk.messaging.session.MessagingSessionStatus

interface WebSocketAdapterObserver {
    fun onConnect()

    fun onMessage(message: String)

    fun onClose(status: MessagingSessionStatus)
}

File: apps/chat-ios/AmazonChimeSDKMessagingDemo/messaging-websocket/internal/WebSocketAdapterObserver.swift
//
//  WebSocketAdapterObserver.swift
//  AmazonChimeSDKMessagingDemo
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: MIT-0
//

public protocol WebSocketAdapterObserver {
    func onConnect()

    func onMessage(message: String)

    func onClose(status: MessagingSessionStatus)
}
